dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/DefaultClients.java

```diff
@@ -151,6 +162,57 @@ private static RemoteException newRemoteException(RemoteException remoteExceptio
         return newException;
     }
 
+    private static RemoteException newCustomRemoteException(RemoteException remoteException) {
+        RemoteException newException = createRemoteExceptionWithReflection(remoteException);
+        newException.initCause(remoteException);
+        return newException;
+    }
+
+    private static RemoteException createRemoteExceptionWithReflection(RemoteException remoteException) {
+        try {
+            Class<?> clazz = remoteException.getClass();
+            boolean foundSerializableErrorProviderInterface = false;
+            // Check if remoteException implements SerializableErrorProvider
+            for (Class<?> classInterface : clazz.getInterfaces()) {
+                if (classInterface != SerializableErrorProvider.class) {
+                    continue;
+                }
+                // Found the interface, now we can proceed to find the constructor
+                foundSerializableErrorProviderInterface = true;
+                break;
+            }
+            if (!foundSerializableErrorProviderInterface) {
+                log.warn("RemoteException subclass does not implement SerializableErrorProvider");
+                return newRemoteException(remoteException);
+            }
```

Comment:
This seems to be simpler?
```suggestion
            Class<?> clazz = remoteException.getClass();
            if (!remoteException instanceOf SerializableErrorProvider) {
                log.warn("RemoteException subclass does not implement SerializableErrorProvider - this is unexpected and will be wrapped and rethrown as a legacy RemoteException", SafeArg.of("class", clazz.getName()));
                return newRemoteException(remoteException);
            }
```